tests/org.eclipse.n4js.spec.tests/xpect-tests/Ch05_05_StructuralTyping/Req76_3_StructuralAndInstanceof.n4js.xt

```diff
@@ -9,21 +9,36 @@
  *   NumberFour AG - Initial API and implementation
  */
 
-class T {
-	public n : number;
-	public foo() {}
+/* XPECT_SETUP org.eclipse.n4js.spec.tests.N4JSSpecTest END_SETUP */
+
+class A {
+    public a: number;
+
+    public f(): void {}
 }
 
-var t : T;
+var a: ~A = { a: 2, f: function () {} };
+let x = 5;
+
+// XPECT errors --> "instanceof cannot be used with use site structural typing." at "~A"
+a instanceof ~A;
+
+// XPECT errors  --> "instanceof cannot be used with use site structural typing." at "~~A"
+a instanceof ~~A;
+
+// The following tests are only to ensure that existing correct functionality is maintained:
+
+// XPECT errors --> "number is not a subtype of union{Function,type{Object},type{N4Enum}}." at "--x"
+a instanceof --x;
+
+// XPECT errors --> "number is not a subtype of union{Function,type{Object},type{N4Enum}}." at "~x"
+a instanceof ~x
+
+// XPECT errors --> "number is not a subtype of union{Function,type{Object},type{N4Enum}}." at "~(3 + 4)"
+a instanceof ~(3 + 4)
 
-/* XPECT errors ---
-"constructor{T} is not a subtype of number." at "T"
-"number is not a subtype of union{Function,type{Object},type{N4Enum}}." at "~T"
---- */
-t instanceof ~T;
+// This needs to work! XPECT noerrors --> 
+a instanceof A;
 
-/* XPECT errors ---
-"constructor{T} is not a subtype of number." at "T"
-"number is not a subtype of union{Function,type{Object},type{N4Enum}}." at "~~T"
---- */
-t instanceof ~~T;
+// XPECT errors --> "instanceof cannot be used with primitive types." at "number" 
+a instanceof number; 
```

Comment:
Maybe add a test with a different unary operator (such as ! or -) and different relational expression such as 

```
a instanceof (-A);
1 < ~A;
```

just to be sure that our extra handling is really limited to ~ and instanceof. (It does at the moment, just in case we will do some stupid refactorings in the future).